src/Factories/Transform/UserSchema/OptionalIfTransformFactory.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using form_builder.Conditions;
+using form_builder.Models;
+using form_builder.Models.Elements;
+
+namespace form_builder.Factories.Transform.UserSchema
+{
+    public class OptionalIfTransformFactory : IUserPageTransformFactory
+    {
+        public async Task<Page> Transform(Page page, FormAnswers convertedAnswers)
+        { 
+            IEnumerable<IElement> elements = page.ValidatableElements
+                .Where(element => !string.IsNullOrEmpty(element.Properties.OptionalIf?.QuestionId));
+
+            if (!elements.Any())
+                return await Task.FromResult(page);
+
+            foreach (IElement element in elements)
+            {
+                var userChoice = convertedAnswers.AllAnswers
```

Comment:
Just a gentle thought... of if you've gone with "IEnumerable<IElement> elements" above.... this would hint at this variable being a "Answers answer"... I think/feel "userChoice" is too specific, as now you're using the Condition the condition could be based on some other factor within the form... blah, blah,blah...... more generic, but hopefully readable